src/base/lane.cc

```diff
@@ -44,8 +44,24 @@ double Lane::do_length() const { return lane_geometry_->ArcLength(); }
 maliput::api::RBounds Lane::do_lane_bounds(double s) const { return lane_geometry_->RBounds(s); }
 
 maliput::api::RBounds Lane::do_segment_bounds(double s) const {
-  // TODO: Implement
-  MALIPUT_THROW_MESSAGE("Not implemented");
+  const double bound_left = ComputeDistanceToSegmentBoundary(kToLeft, s);
+  const double bound_right = ComputeDistanceToSegmentBoundary(kToRight, s);
+  const double tolerance = lane_geometry_->linear_tolerance();
+  return {bound_right < tolerance ? -tolerance : -bound_right, bound_left < tolerance ? tolerance : bound_left};
+}
+
+double Lane::ComputeDistanceToSegmentBoundary(bool to_left, double s) const {
+  const maliput::api::RBounds lane_bounds = do_lane_bounds(s);
+  double lane_bound = to_left ? lane_bounds.max() : -lane_bounds.min();
+  maliput::api::Lane const* other_lane =
+      const_cast<maliput::api::Lane const*>(to_left ? this->to_left() : this->to_right());
```

Comment:
Pff 🤦‍♂️ Sorry it was part of another impl.
Just doing `maliput::api::Lane const* other_lane` is enough. Not needed those cast. 

Small diff: `const maliput::api::Lane* other_lane` won't work because we are replacing that pointer.

Reply:
Done.